recipes/sdsl-lite/all/conanfile.py

```diff
@@ -0,0 +1,43 @@
+import os
+from conans import ConanFile, tools
+
+
+class SDSLLite(ConanFile):
+    name = "sdsl-lite"
+    description = "SDSL - Succinct Data Structure Library"
+    homepage = "https://github.com/xxsds/sdsl-lite"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSD-3-Clause"
+    topics = ("conan", "sdsl", "succint", "data-structures")
+    settings = "compiler"
+    exports = ["patches/*"]
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        source = self.conan_data["sources"][self.version]
+        tools.get(**source)
+        extracted_folder = self.name + "-" + \
+            os.path.splitext(os.path.basename(source["url"]))[0]
+        os.rename(extracted_folder, self._source_subfolder)
+
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
```

Comment:
Since you don't have set `no_copy_source = True`, you can patch the files in the `build` folder.
```suggestion
        def build(self):

        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```

Reply:
i've forgot to set that.
since it's a header only lib, i think the build step would be confusing.
with this in mind, would you like me to do the change?